modules/mini-games/team_production/map_scripts.lua

```diff
@@ -0,0 +1,81 @@
+
+local Public = {}
+
+function Public.create_tiles(distance, tiles, offset_x, offset_y, script_data)
+
+  local count = 1
+  local map_tiles = {}
+  local chunks = {}
+
+  local chunk_distance = math.ceil(distance / 32)
+  local chunk_offset_x = math.floor(offset_x / 32)
+  local chunk_offset_y = math.floor(offset_y / 32)
+  for x = -chunk_distance, chunk_distance do
+    for y = -chunk_distance, chunk_distance do
+      table.insert(chunks, {x = x + chunk_offset_x, y = y + chunk_offset_y})
+    end
+  end
+
+  for name, array_x in pairs (tiles) do
+    for X, array_y in pairs (array_x) do
+      for k, Y in pairs (array_y) do
+        local x = X + offset_x
+        local y = Y + offset_y
+        map_tiles[count] = {name = name, position = {x, y}}
+        count = count + 1
+      end
+    end
+  end
+
+  script_data.surface.set_tiles(map_tiles, true)
+  script_data.surface.regenerate_decorative(nil, chunks)
+end
+
+function Public.recreate_entities(entities, offset_x, offset_y, force, duration, script_data)
```

Comment:
the func has alot lf the same code what i would do is check if the ent is an stack inserter if its not create the ent without direction if it is create it with out. also you can set all other things once and not in all these if statements and use the ifs only for adding them into the table.

Reply:
Not my code so I didnt change it. If it ain't broke, don't fix it.